lvsfunc/deblock.py

```diff
@@ -60,42 +62,46 @@ def autodb_dpir(clip: vs.VideoNode, edgevalue: int = 24,
     except ModuleNotFoundError:
         raise ModuleNotFoundError("autodb_dpir: missing dependency 'vsdpir'")
 
-    assert clip.format
+    if clip.format is None:
+        raise ValueError("autodb_dpir: 'Variable-format clips not supported'")
 
-    def _eval_db(n: int, f: List[vs.VideoFrame], clips: List[vs.VideoNode]) -> vs.VideoNode:
-        out = clips[0]
-        deblocked = False
+    def _eval_db(n: int, f: Sequence[vs.VideoFrame],
+                 clip: vs.VideoNode, db_clips: Sequence[vs.VideoNode],
+                 nthrs: Sequence[Tuple[float, float, float]]) -> vs.VideoNode:
 
-        evref_diff = scale_value(cast(float, f[0].props['EdgeValRefDiff']), 32, 8)
-        y_next_diff = scale_value(cast(float, f[1].props['YNextDiff']), 32, 8)
-        y_prev_diff = scale_value(cast(float, f[2].props['YPrevDiff']), 32, 8)
-        f_type = cast(bytes, f[0].props['_PictType']).decode('utf-8')
+        evref_diff, y_next_diff, y_prev_diff = [
+            get_prop(f[i], prop, float)
+            for i, prop in zip(range(3), ['EdgeValRefDiff', 'YNextDiff', 'YPrevDiff'])
+        ]
+        f_type = get_prop(f[0], '_PictType', bytes).decode('utf-8')
 
         if f_type == 'I':
             y_next_diff = (y_next_diff + evref_diff) / 2
 
-        for i, thr in enumerate(thrs):
-            if evref_diff > thr[0] and y_next_diff > thr[1] and y_prev_diff > thr[2]:
-                out, i = clips[i+1], i
-                deblocked = True
-
-        out = out.std.SetFrameProp('adb_EdgeValRefDiff', floatval=evref_diff) \
-            .std.SetFrameProp('adb_YNextDiff', floatval=y_next_diff) \
-            .std.SetFrameProp('adb_YPrevDiff', floatval=y_prev_diff)
-
-        if deblocked:
-            # TO-DO: Figure out why it doesn't appear to be adding these props
-            out.std.SetFrameProp('adb_EdgeValRefDiff_threshold', floatval=thrs[i][0]) \
-                .std.SetFrameProp('adb_YNextDiff_threshold', floatval=thrs[i][1]) \
-                .std.SetFrameProp('adb_YPrevDiff_threshold', floatval=thrs[i][2])
+        out = clip
+        nthr_used = (-1., ) * 3
+        for dblk, nthr in zip(db_clips, nthrs):
+            if all(p > t for p, t in zip([evref_diff, y_next_diff, y_prev_diff], nthr)):
+                out = dblk
+                nthr_used = nthr
+
+        if write_props:
+            for prop_name, prop_val in zip(
+                ['AdbEdgeValRefDiff', 'AdbYNextDiff', 'AdbYPrevDiff',
+                 'AdbEdgeValRefDiffThreshold', 'AdbYNextDiffThreshold', 'AdbYPrevDiffThreshold'],
+                [evref_diff, y_next_diff, y_prev_diff] + list(nthr_used)
+            ):
+                out = out.std.SetFrameProp(prop_name, floatval=prop_val)
```

Comment:
The output here is different from the thresholds passed in the params, which can lead to confusion. I'd prefer them being converted to whatever thresholds we're sticking with. I'm leaning more towards a [0, 100] range currently, but I'm not entirely sure yet. Could also leave this for a future commit if we can't decide right now.